### PR TITLE
gurk-rs: 0.4.3 -> 0.5.1

### PR DESCRIPTION
**Pull request by devhell**
## Description of changes

Related https://github.com/NixOS/nixpkgs/issues/332957

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/gurk-rs/default.nix [hunk 1]
```diff
@@ -25,22 +27,25 @@ rustPlatform.buildRustPackage rec {
...
+      "libsignal-protocol-0.1.0" = "sha256-4aHINlpVAqVTtm7npwXQRutZUmIxYgkhXhApg7jSM4M=";
+      "libsignal-service-0.1.0" = "sha256-AOGw76A9R2qH3hc7B+MBE3okzW8b5LTZdepzUDOv9lM=";
+      "curve25519-dalek-4.1.3" = "sha256-bPh7eEgcZnq9C3wmSnnYv0C4aAP+7pnwk9Io29GrI4A=";
+      "presage-0.6.2" = "sha256-t9t8ecPtefI/jYlk+Ul8WdgH26VJIkfMptbKxprekS0=";
       "qr2term-0.3.1" = "sha256-U8YLouVZTtDwsvzZiO6YB4Pe75RXGkZXOxHCQcCOyT8=";
     };
   };
 
-  nativeBuildInputs = [ protobuf ];
+  nativeBuildInputs = [ protobuf perl ];
```
Comment (alyssais) [inline pkgs/applications/networking/instant-messengers/gurk-rs/default.nix:38]: Is this for OpenSSL? If so, it's better to set `OPENSSL_NO_VENDOR = true`, to use the version from Nixpkgs.

File: pkgs/applications/networking/instant-messengers/gurk-rs/default.nix [hunk 2]
```diff
@@ -25,22 +28,27 @@ rustPlatform.buildRustPackage rec {
...
+      "libsignal-protocol-0.1.0" = "sha256-4aHINlpVAqVTtm7npwXQRutZUmIxYgkhXhApg7jSM4M=";
+      "libsignal-service-0.1.0" = "sha256-AOGw76A9R2qH3hc7B+MBE3okzW8b5LTZdepzUDOv9lM=";
+      "curve25519-dalek-4.1.3" = "sha256-bPh7eEgcZnq9C3wmSnnYv0C4aAP+7pnwk9Io29GrI4A=";
+      "presage-0.6.2" = "sha256-t9t8ecPtefI/jYlk+Ul8WdgH26VJIkfMptbKxprekS0=";
       "qr2term-0.3.1" = "sha256-U8YLouVZTtDwsvzZiO6YB4Pe75RXGkZXOxHCQcCOyT8=";
     };
   };
 
-  nativeBuildInputs = [ protobuf ];
+  nativeBuildInputs = [ protobuf perl pkg-config ];
```
Comment (Aleksanaa) [inline pkgs/applications/networking/instant-messengers/gurk-rs/default.nix:39]: Still needing perl?

Comment (devhell) [reply pkgs/applications/networking/instant-messengers/gurk-rs/default.nix:39]: Sorry, brainfart, will fix in a sec.

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 340311` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

**Comment by Scrumplex** [comment]
I tried to link my device but I got the following error:

```
Error: failed to provision device: Service error: Unexpected response: HTTP 409

Caused by:
    0: Service error: Unexpected response: HTTP 409
    1: Unexpected response: HTTP 409
```

**Comment by devhell** [comment]
> I tried to link my device but I got the following error:
> 
> ```
> Error: failed to provision device: Service error: Unexpected response: HTTP 409
> 
> Caused by:
>     0: Service error: Unexpected response: HTTP 409
>     1: Unexpected response: HTTP 409
> ```

This was fixed in 0.5.1. Are you sure you're executing the built binary and no the old 0.4.3 binary?

**Comment by Scrumplex** [comment]
> This was fixed in 0.5.1. Are you sure you're executing the built binary and not the old 0.4.3 binary?

I was running it in the shell generated by nixpkgs-review, I don't have `gurk` in my PATH outside of it :O

I have entered the nixpkgs-review shell again and `gurk -V` yields `gurk 0.5.1`

**Comment by devhell** [comment]
> > This was fixed in 0.5.1. Are you sure you're executing the built binary and not the old 0.4.3 binary?
> 
> I was running it in the shell generated by nixpkgs-review, I don't have `gurk` in my PATH outside of it :O
> 
> I have entered the nixpkgs-review shell again and `gurk -V` yields `gurk 0.5.1`
> 
> Edit:
> 
> debug log output:
> 
> ```
> 2024-09-07T21:32:56.952451Z  INFO presage::manager::linking: generating qrcode from provisioning link: sgnl://linkdevice?uuid=[redacted]&pub_key=[redacted]    
> 2024-09-07T21:33:07.637036Z  WARN libsignal_service_hyper::websocket: Websocket closing: Some(CloseFrame { code: Normal, reason: "Closed" })
> 2024-09-07T21:33:07.637085Z ERROR libsignal_service::websocket: SignalWebSocket: Websocket error: end of web request stream; socket closing
> 2024-09-07T21:33:07.923850Z ERROR put_json{service=Service path="/v1/devices/link" additional_headers=[] credentials_override=Identified(HttpAuth { username: "[redacted]" })}:request{method=PUT endpoint=Service additional_headers=[] credentials_override=Identified(HttpAuth { username: "[redacted]" }) path=/v1/devices/link}: libsignal_service_hyper::push_service: Failed to decode HTTP 409 response: Error decoding JSON response: missing field `missingDevices` at line 1 column 42
> ```

Hmmm, ok, I have had that issue with 0.4.3, and upstream fixed it apparently in this version. I can't reproduce your issue on my side. I can try to have another go at it tomorrow. I think the best way to proceed would be to perhaps open an issue with upstream for your case? Sorry about this.

**Comment by devhell** [comment]
@Scrumplex, so I've tested it again. Removed `~/.config/gurk` and started version `0.5.1` again from fresh, which created a QR code, which I then scanned, and it linked successfully. The issue upstream fixed is described in this issue: https://github.com/boxdot/gurk-rs/issues/298

**Comment by devhell** [comment]
@alyssais can this be merged now, please?

**Review by Scrumplex — APPROVED** [review]
Packaging changes LGTM

**Pull request merged**